pkg/operator/client/deploy.go

```diff
@@ -481,6 +584,52 @@ func getRemovedCharts(prevDir string, curDir string, previousKotsCharts []v1beta
 	return removedCharts, nil
 }
 
+func findMatchingHelmCharts(chartsDir string, kotsCharts []kotsutil.HelmChartInterface) ([]kotsutil.HelmChartInterface, error) {
+	dirContent, err := ioutil.ReadDir(chartsDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list chart dir %s", chartsDir)
+	}
+
+	matching := []kotsutil.HelmChartInterface{}
+
+	for _, kotsChart := range kotsCharts {
+		for _, f := range dirContent {
+			if !f.IsDir() {
+				continue
+			}
+
+			dirName := f.Name()
+			chartDir := filepath.Join(chartsDir, dirName)
+
+			var chartName, chartVersion string
+			if kotsChart.GetAPIVersion() == "kots.io/v1beta1" {
+				chartName, chartVersion, err = findChartNameAndVersion(chartDir)
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to find chart name and version in %s", chartDir)
+				}
+			} else if kotsChart.GetAPIVersion() == "kots.io/v1beta2" {
+				archivePath, err := findChartTgz(chartDir)
```

Comment:
can you add a comment here that v1beta2 charts are tgz? maybe add a comment to the above block for v1beta1 that they're not tgz and already extracted?